lib/rubocop/cop/rspec/file_path.rb

```diff
@@ -82,30 +82,39 @@ def on_top_level_example_group(node)
         private
 
         def ensure_correct_file_path(send_node, described_class, arguments)
-          glob = glob_for(described_class, arguments.first)
-          return if filename_ends_with?(glob)
-
-          add_offense(send_node, message: format(MSG, suffix: glob))
+          pattern = pattern_for(described_class, arguments.first)
+          return if filename_ends_with?(pattern)
+
+          # For the suffix shown in the offense message, modify the regular
+          # expression pattern to resemble a glob pattern for clearer error
+          # messages.
+          offense_suffix = pattern.gsub('.*', '*').sub('[^/]', '')
+            .sub('\.', '.')
+          add_offense(send_node, message: format(MSG, suffix: offense_suffix))
         end
 
         def routing_spec?(args)
           args.any?(&method(:routing_metadata?))
         end
 
-        def glob_for(described_class, method_name)
-          return glob_for_spec_suffix_only? if spec_suffix_only?
+        def pattern_for(described_class, method_name)
+          return pattern_for_spec_suffix_only? if spec_suffix_only?
 
-          "#{expected_path(described_class)}#{name_glob(method_name)}*_spec.rb"
+          [
+            expected_path(described_class),
+            name_pattern(method_name),
+            '[^/]*_spec\.rb'
+          ].join
         end
 
-        def glob_for_spec_suffix_only?
-          '*_spec.rb'
+        def pattern_for_spec_suffix_only?
+          '.*_spec.rb'
         end
 
-        def name_glob(method_name)
+        def name_pattern(method_name)
           return unless method_name&.str_type?
 
-          "*#{method_name.str_content.gsub(/\W/, '')}" unless ignore_methods?
+          ".*#{method_name.str_content.gsub(/\W/, '')}" unless ignore_methods?
```

Comment:
I don't exactly agree with the following offence:
```
/Users/pirj/source/rspec-dev/repos/rspec-expectations/spec/rspec/expectations/failure_aggregator_spec.rb:2:3: C: RSpec/FilePath: Spec path should end with failure_aggregator*whenusedviaaggregate_failures*_spec.rb.
  RSpec.describe FailureAggregator, "when used via `aggregate_failures`" do
  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
As I would expect `whenusedviaaggregate_failures` to be more like `when_used_via_aggregate_failures`, but it has nothing to do with this pull request.

Reply:
I haven’t double checked this problem, but could you open it as a new issue @pirj ?

Reply:
Filed all three deficiencies as tickets 👍